python/ray/serve/schema.py

```diff
@@ -487,6 +517,118 @@ def get_empty_schema_dict() -> Dict:
+    deployment_status: DeploymentStatus = Field(
+        description="The current status of the deployment."
+    )
+    message: str = Field(
+        description=(
+            "If the deployment status is UNHEALTHY, this will describe the issue in "
+            "more detail."
+        )
+    )
+    deployment_config: DeploymentSchema = Field(
+        description=(
+            "The set of deployment config options that are currently applied to this "
+            "deployment. These options may come from users code, or from override "
+            "options specified in the config file applied to the cluster, or could be "
+            "default values loaded in if unspecified."
+        )
+    )
+
+    @validator("deployment_config")
+    def deployment_route_prefix_not_set(cls, v: DeploymentSchema):
+        # Route prefix should not be set at the deployment level. Deployment-level route
+        # prefix is outdated, there should be one route prefix per application
+        if "route_prefix" in v.dict(exclude_unset=True):
+            raise ValueError(
+                "Unexpectedly found a non-null route_prefix in the deployment_config "
+                f'for deployment "{cls.name}". The route_prefix in deployment_config '
+                "within DeploymentDetails should not be set; please set it at the "
+                "application level."
+            )
+        return v
+
+
+@PublicAPI(stability="alpha")
+class ApplicationDetails(BaseModel, extra=Extra.forbid):
+    name: str = Field(description="Application name.")
+    route_prefix: Optional[str] = Field(
+        ...,
+        description=(
+            "Requests to paths under this HTTP path prefix will be routed to this "
+            "application. This value can be null if the application is deploying "
+            "and app information has not been fully propagated in the backend; or "
+            "if the application is not exposed over HTTP. Routing is done based on "
+            'longest-prefix match, so if you have deployment A with a prefix of "/a" '
+            'and deployment B with a prefix of "/a/b", requests to "/a", "/a/", and '
+            '"/a/c" go to A and requests to "/a/b", "/a/b/", and "/a/b/c" go to B. '
+            'Routes must not end with a "/" unless they\'re the root (just "/"), which '
+            "acts as a catch-all."
+        ),
+    )
+    docs_path: Optional[str] = Field(
+        ...,
+        description=(
+            "The path at which the docs for this application is served, for instance "
+            "the `docs_url` for FastAPI-integrated applications."
+        ),
+    )
+    app_status: ApplicationStatus = Field(
+        description="The current status of the application."
+    )
+    app_message: str = Field(
+        description="A message that gives more insight into the application status."
+    )
+    deployment_timestamp: float = Field(
+        description="The time at which the application was deployed"
```

Comment:
can we describe if this is seconds or milleseconds?
in state-api i see fields are named like this: `start_time_ms`

Reply:
Thanks for the suggestion! Changed to `last_deployed_s` from what I remember of our sync with @edoakes, please suggest if you think there is a more intuitive name.